crates/encoding/src/config.rs

```diff
@@ -0,0 +1,327 @@
+pub struct BufferSizes {
+    // Known size buffers
+    pub path_reduced: BufferSize<PathMonoid>,
+    pub path_reduced2: BufferSize<PathMonoid>,
+    pub path_reduced_scan: BufferSize<PathMonoid>,
+    pub path_monoids: BufferSize<PathMonoid>,
+    pub path_bboxes: BufferSize<PathBbox>,
+    pub cubics: BufferSize<Cubic>,
+    pub draw_reduced: BufferSize<DrawMonoid>,
+    pub draw_monoids: BufferSize<DrawMonoid>,
+    pub info: BufferSize<u32>,
+    pub clip_inps: BufferSize<Clip>,
+    pub clip_els: BufferSize<ClipElement>,
+    pub clip_bics: BufferSize<ClipBic>,
+    pub clip_bboxes: BufferSize<ClipBbox>,
+    pub draw_bboxes: BufferSize<DrawBbox>,
+    pub bump_alloc: BufferSize<BumpAllocators>,
+    pub bin_headers: BufferSize<BinHeader>,
+    pub paths: BufferSize<Path>,
+    // Bump allocated buffers
+    pub bin_data: BufferSize<u32>,
+    pub tiles: BufferSize<Tile>,
+    pub segments: BufferSize<PathSegment>,
+    pub ptcl: BufferSize<u32>,
+}
+
+impl BufferSizes {
+    pub fn new(layout: &Layout, workgroups: &WorkgroupCounts, n_path_tags: u32) -> Self {
+        let n_paths = layout.n_paths;
+        let n_draw_objects = layout.n_draw_objects;
+        let n_clips = layout.n_clips;
+        let path_tag_wgs = workgroups.path_reduce.0;
+        let reduced_size = if workgroups.use_large_path_scan {
+            align_up(path_tag_wgs, PATH_REDUCE_WG)
+        } else {
+            path_tag_wgs
+        };
+        let path_reduced = BufferSize::new(reduced_size);
+        let path_reduced2 = BufferSize::new(PATH_REDUCE_WG);
+        let path_reduced_scan = BufferSize::new(path_tag_wgs);
+        let path_monoids = BufferSize::new(path_tag_wgs * PATH_REDUCE_WG);
+        let path_bboxes = BufferSize::new(n_paths);
+        let cubics = BufferSize::new(n_path_tags);
+        let draw_object_wgs = workgroups.draw_reduce.0;
+        let draw_reduced = BufferSize::new(draw_object_wgs);
+        let draw_monoids = BufferSize::new(n_draw_objects);
+        let info = BufferSize::new(layout.bin_data_start);
+        let clip_inps = BufferSize::new(n_clips);
+        let clip_els = BufferSize::new(n_clips);
+        let clip_bics = BufferSize::new(n_clips / CLIP_REDUCE_WG);
+        let clip_bboxes = BufferSize::new(n_clips);
+        let draw_bboxes = BufferSize::new(n_paths);
+        let bump_alloc = BufferSize::new(1);
+        let bin_headers = BufferSize::new(draw_object_wgs * 256);
+        let n_paths_aligned = align_up(n_paths, 256);
+        let paths = BufferSize::new(n_paths_aligned);
+
+        // The following buffer sizes have been hand picked to accommodate the vello test scenes as
+        // well as paris-30k. These should instead get derived from the scene layout using
+        // reasonable heuristics.
+        let bin_data = BufferSize::from_size_in_bytes(1 << 20);
+        let tiles = BufferSize::from_size_in_bytes(1 << 24);
+        let segments = BufferSize::from_size_in_bytes(1 << 26);
```

Comment:
In the original code these, were `/ TILE_SIZE` and `/ SEGMENT_SIZE` respectively. Not too important, but we should probably use `new` instead of `from_size_in_bytes` here to keep the sizes the same.

Reply:
In the original code these size were multiplied by`TILE_SIZE` and `SEGMENT_SIZE` later for buffer allocation, so I think they were given in bytes. That said there is no reason why these couldn't use `::new`. I changed these numbers to be in terms of element counts instead of bytes. The segments buffer is now smaller than this number but it seems to still accommodate paris-30k on my machine.